hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -237,105 +223,60 @@ func MustAsset(name string) string {
 }
 
 var (
-	setupAPIServerIPScriptTemplate       = template.Must(template.New("setupAPIServerIP").Parse(MustAsset("apiserver-haproxy/setup-apiserver-ip.sh")))
-	teardownAPIServerIPScriptTemplate    = template.Must(template.New("teardownAPIServerIP").Parse(MustAsset("apiserver-haproxy/teardown-apiserver-ip.sh")))
-	haProxyConfigTemplate                = template.Must(template.New("haProxyConfig").Parse(MustAsset("apiserver-haproxy/haproxy.cfg")))
-	setupRemoteKASSVCIPScriptTemplate    = template.Must(template.New("setupRemoteKASSVCIPScript").Parse(MustAsset("remote-kas-svc-haproxy/setup-remote-kas-svc-ip.sh")))
-	teardownRemoteKASSVCIPScriptTemplate = template.Must(template.New("teardownRemoteKASSVCIPScript").Parse(MustAsset("remote-kas-svc-haproxy/teardown-remote-kas-svc-ip.sh")))
-	haProxyConfigRemoteKASSVCTemplate    = template.Must(template.New("haProxyConfigRemoteKASSVC").Parse(MustAsset("remote-kas-svc-haproxy/haproxy.cfg")))
+	setupAPIServerIPScriptTemplate    = template.Must(template.New("setupAPIServerIP").Parse(MustAsset("apiserver-haproxy/setup-apiserver-ip.sh")))
+	teardownAPIServerIPScriptTemplate = template.Must(template.New("teardownAPIServerIP").Parse(MustAsset("apiserver-haproxy/teardown-apiserver-ip.sh")))
+	haProxyConfigTemplate             = template.Must(template.New("haProxyConfig").Parse(MustAsset("apiserver-haproxy/haproxy.cfg")))
 )
 
-func apiServerProxyConfig(haProxyImage, cpoImage,
-	externalAPIAddress, internalAPIAddress, remoteKASSVCAdress string,
+func apiServerProxyConfig(haProxyImage, cpoImage, clusterID,
+	externalAPIAddress, internalAPIAddress string,
 	externalAPIPort, internalAPIPort int32, proxyAddr, noProxy, serviceNetwork, clusterNetwork string) ([]byte, error) {
 	config := &ignitionapi.Config{}
 	config.Ignition.Version = ignitionapi.MaxVersion.String()
 
+	if sharedingress.UseSharedIngress() {
+		// TODO: get the image from somewhere else
+		// proxy protocol v2 requires haproxy v2.9+
+		haProxyImage = "docker.io/haproxy:2.9.9"
```

Comment:
wouldn't we hit docker pull rate limiting e.g in CI? can we please source this from quay?

Also we were using proxy protocol already, so I assume you mean TLV support?
Can add a link in the comment backing the statement, e.g. https://www.haproxy.com/blog/announcing-haproxy-2-9#proxy-protocol-tlv-fields

Also I created https://issues.redhat.com/browse/HOSTEDCP-1819 to track this.

Reply:
we were using proxy protocol v1, not v2. But yeah, TLV support is what we need. I will update it.

Reply:
done